js/pages/configuration/configuration.js

```diff
@@ -127,29 +163,52 @@ define([
       this.updateSourceDaimonPriority(selectedSource.sourceKey, 'Results');
       return true;
     };
-    
+
     checkSourceConnection(source) {
       sourceApi.checkSourceConnection(source.sourceKey)
         .then( ({ data }) =>
-           source.connectionCheck(data.sourceId === undefined ?
-               sourceApi.connectionCheckState.failed : sourceApi.connectionCheckState.success))
-        .catch(() => {source.connectionCheck(sourceApi.connectionCheckState.failed);});
-        source.connectionCheck(sourceApi.connectionCheckState.checking);
+          source.connectionCheck(data.sourceId === undefined ?
+            sourceApi.buttonCheckState.failed : sourceApi.buttonCheckState.success))
+        .catch(() => {source.connectionCheck(sourceApi.buttonCheckState.failed);});
+      source.connectionCheck(sourceApi.buttonCheckState.checking);
     };
-    
+
+    async refreshSourceCache(source) {
+      try {
+        const { data } = await sourceApi.refreshSourceCache(source.sourceKey);
+        if(data.sourceId === undefined) {
```

Comment:
Where is `sourceId` in the response?
![image](https://user-images.githubusercontent.com/10010071/55025085-9309cf00-4fd6-11e9-846a-5e0e8fcb60bf.png)